src/Exporter.php

```diff
@@ -707,9 +707,17 @@ protected function getRequiredVariationValues():array
             $variationValues[] = 'variationAttributeValues';
         }
 
+        $itemProperties = $this->registry->get('itemproperties');
         $properties = $this->registry->get('properties');
-        if ($properties && !empty($properties->getResults())) {
-            $variationValues[] = 'variationProperties';
+
+        if ($itemProperties && $properties) {
+
+            // If there are no properties at all, request no properties as we do not really need them and they
+            // take a long time to fetch from the API.
+            $allProperties = array_merge($itemProperties->getResults(), $properties->getResults());
+            if ($allProperties && !empty($allProperties)) {
```

Comment:
> Returns the resulting array. If called without any arguments, returns an empty array. 

`array_merge` shouldn't return `null`, so checking for `empty` should be enough?!